Content.Shared/Prototypes/LocalizedPrototype.cs

```diff
@@ -0,0 +1,37 @@
+using System.Linq;
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Prototypes;
+
+public abstract class LocalizedPrototype : IPrototype
+{
+    [IdDataField]
+    public string ID { get; } = default!;
+
+    public const string LocFormat = "{type}-{ID}-{field}";
+
+    /// <summary>The localization string for the name of this prototype</summary>
+    public string NameLoc => ToLocalizationString("name");
+    /// <summary>The localized string for the name of prototype</summary>
+    public string Name => Loc.GetString(NameLoc);
+
+    public string ToLocalizationString(string field)
```

Comment:
```suggestion
    /// <summary>
    ///     Returns an Loc string using the input as the 'property.
    ///     Given `desc` it will return `ThisPrototype-PrototypeId-desc`.
    ///     These strings can be used to automatically localize Prototypes.
    /// </summary>
    public string PropertyLocString(string field)
```